torch_xla/distributed/spmd/xla_sharding.py

```diff
@@ -584,6 +584,12 @@ def mark_sharding(t: Union[torch.Tensor, XLAShardedTensor], mesh: Mesh,
   assert len(t.shape) == len(partition_spec), \
     f"Partition spec length ({len(partition_spec)}) should be equal to the input rank ({len(t.shape)})."
 
+  tx = maybe_get_torchax()
```

Comment:
Can we add a test for this case? I think maybe running a graph with `assume_pure(mark_sharding)` and then testing that the outputs were sharded correctly like the `test_scan_spmd.py` are doing

Reply:
done